rosbag2_storage_mcap/src/mcap_storage.cpp

```diff
@@ -735,6 +741,7 @@ uint64_t MCAPStorage::get_minimum_split_file_size() const
 /** BaseWriteInterface **/
 void MCAPStorage::write(std::shared_ptr<const rosbag2_storage::SerializedBagMessage> msg)
 {
+  std::shared_lock read_lock(mcap_rw_shared_mutex_);
```

Comment:
@fujitatomoya Actually the `shared_mutex` in this particular case has the opposite effect on performance and will be on average **4.5 times slower** than regular `std::mutex` with `unique_lock` because we don't have cases in rosbag2 when `MCAPStorage::write(std::shared_ptr<const rosbag2_storage::SerializedBagMessage> msg)` could be called from multiple threads.

Reply:
@MichaelOrlov what if we have MultiThreadedExecutor? i was expecting that will call backend storage `write` call concurrently? i may be mistaken...